specs/account_spec.rb

```diff
@@ -137,35 +159,72 @@
 end
 
 # TODO: change 'xdescribe' to 'describe' to run these tests
-xdescribe "Wave 2" do
+describe "Wave 2" do
+
+  before do
+    @id =[]
+    @balance =[]
+    @open_date =[]
+
+    CSV.open("../support/accounts.csv").each do |line|
+      @id << line[0].to_i
+      @balance << line[1].to_i
+      @open_date << line[2]
+    end
+
+  end
+
   describe "Account.all" do
+
     it "Returns an array of all accounts" do
       # TODO: Your test code here!
       # Useful checks might include:
       #   - Account.all returns an array
+      Bank::Account.all.class.must_equal Array
+
       #   - Everything in the array is an Account
       #   - The number of accounts is correct
-      #   - The ID and balance of the first and last
-      #       accounts match what's in the CSV file
+      number_of_accounts = 0
+      CSV.read("/Users/adai/Documents/ada/projects/BankAccounts/support/accounts.csv").each do |line|
+        number_of_accounts += 1
+        return number_of_accounts
+      end
+
+      Bank::Account.all.length.must_be number_of_accounts
+
+      Account.all[0][0].must_equal @id[0] #first ID
+      Account.all[0][1].must_equal @balance[0] #first balance
+      Account.all[-1][0].must_equal @id[-1] #last ID
+      Account.all[-1][1].must_equal @balance[-1] #last Balance
+
+
       # Feel free to split this into multiple tests if needed
     end
   end
 
+
   describe "Account.find" do
     it "Returns an account that exists" do
       # TODO: Your test code here!
+      Bank::Account.find(@id[5]).must_be_instance_of Bank::Account
     end
 
     it "Can find the first account from the CSV" do
       # TODO: Your test code here!
+      Bank::Account.find(@id[0]).must_be_instance_of Bank::Account
```

Comment:
More than it just finding something, it should also make sure that the balance, etc are correct.